File: opcua-webservice-spring/examples/uawebservicedemo/src/main/java/org/opcfoundation/uawebservicedemo/models/company/department/DepartmentEmployeeReferenceType.java
package org.opcfoundation.uawebservicedemo.models.company.department;

import org.eclipse.milo.opcua.stack.core.StatusCodes;
import org.eclipse.milo.opcua.stack.core.UaRuntimeException;
import org.eclipse.milo.opcua.stack.core.types.builtin.LocalizedText;
import org.opcfoundation.uawebservicedemo.database.EmployeeDatabase;
import org.opcfoundation.uawebservicedemo.database.entity.Department;
import org.opcfoundation.uawebservicedemo.database.entity.Employee;
import org.opcfoundation.uawebservicedemo.database.mapper.DepartmentMapper;
import org.opcfoundation.uawebservicedemo.database.mapper.EmployeeMapper;
import org.opcfoundation.uawebservicedemo.models.EmployeeTwinSpace;
import org.opcfoundation.uawebservicedemo.models.employee.EmployeeDigitalTwinType;
import org.opcfoundation.webserver.digitaltwin.element.ReferenceElementType;
import org.opcfoundation.webserver.service.message.digitaltwin.GetLinkRequest;
import org.opcfoundation.webserver.service.message.digitaltwin.GetLinkResponse;
import org.opcfoundation.webserver.types.digitaltwin.ReferenceTargetDescriptor;

import java.util.List;
import java.util.concurrent.CompletableFuture;

// Defining employee list of department
public class DepartmentEmployeeReferenceType extends ReferenceElementType {
    public DepartmentEmployeeReferenceType(EmployeeTwinSpace twinSpace) {
        super("DepartmentEmployeeReferenceType", new LocalizedText("DepartmentEmployee"), twinSpace);
    }

    // Get employee of department
    @Override
    public CompletableFuture<GetLinkResponse> onGetLinks(GetLinkRequest request)
    {
        try
        {
            // Get department id from request
            Integer departmentId = Integer.parseInt(request.getId());

            // Database query
            EmployeeMapper employeeMapper = EmployeeDatabase.getEmployeeMapper();
            List<Employee> employees = employeeMapper.getEmployeeOfDepartment(departmentId);

            // Return department general information
            GetLinkResponse response = new GetLinkResponse();
            EmployeeDigitalTwinType employeeDigitalTwinType = ((EmployeeTwinSpace)nodeManager).employeeDigitalTwinType;

            for (Employee item : employees)
            {
                ReferenceTargetDescriptor descriptor = new ReferenceTargetDescriptor(
                        item.getId().toString(), // EmployeeId
                        new LocalizedText(item.getName()), // EmployeeName
                        employeeDigitalTwinType); // EmployeeDigitalTwinType

                response.add(descriptor);
            }

            return CompletableFuture.completedFuture(response);

        } catch (Exception e) {
            throw new UaRuntimeException(StatusCodes.Bad_NodeIdUnknown);
        }
    }
}

File: opcua-webservice-node/examples/demoserver/src/models/company/department/DepartmentEmployeeReferenceType.ts
import { StatusCodes } from "opcua-webapi";
import { UaError, UaLocalizedText } from "opcua-webapi-ts";
import {
    GetLinkRequest,
    GetLinkResponse,
    ReferenceElementType,
    ReferenceTargetDescriptor
} from "opcua-webservice-node";

import { prisma } from "../../../connectors/prismaClient";
import { EmployeeTwinSpace } from "../../EmployeeTwinSpace";

export class DepartmentEmployeeReferenceType extends ReferenceElementType {
    constructor(twinSpace: EmployeeTwinSpace) {
        super("DepartmentEmployeeReferenceType", new UaLocalizedText("DepartmentEmployee"), twinSpace);
    }

    override async onGetLinks(request: GetLinkRequest): Promise<GetLinkResponse> {
        // Get department id from request
        const departmentId = Number.parseInt(request.id, 10);
        if (Number.isNaN(departmentId)) {
            throw UaError.from(StatusCodes.BadNodeIdUnknown);
        }

        // Database query
        let department = await prisma.department.findFirst({
            where: { ID: departmentId },
            select: {                    
                Name: true,
                employeedepartment: { 
                    select: { employee: { select: { ID: true, Name: true } } } }
                }
        });

        if (department === null) {
            throw UaError.from(StatusCodes.BadNodeIdUnknown);
        }

        console.log(department.employeedepartment);

        let response = new GetLinkResponse();

        for (const item of department.employeedepartment) {
            const descriptor = new ReferenceTargetDescriptor(
                item.employee.ID.toString(), 
                new UaLocalizedText(item.employee.Name), 
                EmployeeTwinSpace.EmployeeDigitalTwinType);
            response.add(descriptor);
        }

        return response;   
    }
}